File: mobile_flutter/lib/views/detail_view.dart
// ============================================================
// [Software Eng] Detail View — OLED Dark + Glass Morphic
// ============================================================
// Per-post breakdown with frosted glass cards for each modality.
// LIME explanation, decision path, Grad-CAM placeholder.

import 'package:flutter/material.dart';
import '../theme/monochrome_glass.dart';
import '../components/glass_card.dart';
import '../components/confidence_gauge.dart';

class DetailView extends StatefulWidget {
  final Map<String, dynamic> post;
  const DetailView({super.key, required this.post});

  @override
  State<DetailView> createState() => _DetailViewState();
}

class _DetailViewState extends State<DetailView> {
  int _selectedSegment = 0;

  @override
  Widget build(BuildContext context) {
    final score = widget.post['score'] as double;
    final dark = Theme.of(context).brightness == Brightness.dark;
    return Scaffold(
      appBar: AppBar(title: const Text('Analysis')),
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(HIGSpacing.md),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text(widget.post['title'] as String,
                  style: Theme.of(context).textTheme.displayMedium),
              const SizedBox(height: HIGSpacing.xs),
              Row(
                children: [
                  Icon(Icons.schedule, size: 14,
                      color: dark ? DarkColors.tertiaryLabel : LightColors.tertiary),
                  const SizedBox(width: 4),
                  Text('Just now', style: Theme.of(context).textTheme.bodyMedium),
                  const Spacer(),
                  Icon(Icons.person_outline, size: 14,
                      color: dark ? DarkColors.tertiaryLabel : LightColors.tertiary),
                  const SizedBox(width: 4),
                  Text(widget.post['source'] as String,
                      style: Theme.of(context).textTheme.bodyMedium),
                ],
              ),
              const SizedBox(height: HIGSpacing.lg),
              GlassCard(
                title: 'Overall Suspicion Score',
                child: Column(
                  children: [
                    Center(
                      child: ConfidenceGauge(
                        score: score,
                        label: score > 0.7 ? 'Likely Misinformation' : 'Likely Authentic',
                        size: 120,
                      ),
                    ),
                    const SizedBox(height: HIGSpacing.md),
                    ConfidenceBar(score: score, label: 'Confidence'),
                  ],
                ),
              ),
              const SizedBox(height: HIGSpacing.md),
              _segmentPicker(dark),
              const SizedBox(height: HIGSpacing.md),
              _segmentContent(dark),
            ],
          ),
        ),
      ),
    );
  }

  Widget _segmentPicker(bool dark) {
    final labels = ['Overview', 'Modalities', 'Explain'];
    return Container(
      decoration: BoxDecoration(
        color: dark ? DarkColors.glassBg : LightColors.glassBg,
        borderRadius: BorderRadius.circular(10),
        border: Border.all(
            color: dark ? DarkColors.glassBorder : LightColors.glassBorder),
      ),
      child: Row(
        children: labels.asMap().entries.map((entry) {
          final i = entry.key;
          final label = entry.value;
          final isSelected = _selectedSegment == i;
          return Expanded(
            child: GestureDetector(
              onTap: () => setState(() => _selectedSegment = i),
              child: Container(
                padding: const EdgeInsets.symmetric(vertical: 10),
                decoration: BoxDecoration(
                  color: isSelected
                      ? (dark ? DarkColors.primary : LightColors.primary)
                      : Colors.transparent,
                  borderRadius: BorderRadius.circular(8),
                ),
                child: Text(label,
                    textAlign: TextAlign.center,
                    style: TextStyle(
                      fontSize: 13,
                      fontWeight: FontWeight.w600,
                      color: isSelected
                          ? (dark ? DarkColors.background : LightColors.background)
                          : (dark ? DarkColors.secondaryLabel : LightColors.secondaryLabel),
                    )),
              ),
            ),
          );
        }).toList(),
      ),
    );
  }

  Widget _segmentContent(bool dark) {
    switch (_selectedSegment) {
      case 0: return _overviewContent(dark);
      case 1: return _modalitiesContent(dark);
      case 2: return _explainContent(dark);
      default: return const SizedBox.shrink();
    }
  }

  Widget _overviewContent(bool dark) {
    return Column(
      children: [
        _overviewCard('Text Analysis', 'Semantic and linguistic patterns',
            Icons.text_fields, 'Normal', 0.2, 'BERT-based encoding'),
        const SizedBox(height: HIGSpacing.md),
        _overviewCard('Image Analysis', 'Visual integrity via ResNet',
            Icons.photo, 'Authentic', 0.15, 'No manipulation detected'),
        const SizedBox(height: HIGSpacing.md),
        GlassCard(title: 'Source Credibility',
            subtitle: 'Dynamic scoring based on behavior',
            icon: Icons.verified_outlined,
            child: ConfidenceBar(score: 0.78, label: 'Credibility Score')),
        const SizedBox(height: HIGSpacing.md),
        GlassCard(title: 'Propagation Pattern',
            subtitle: 'Share network modeled as graph',
            icon: Icons.account_tree_outlined,
            child: Row(
              children: [
                SuspicionBadge(level: 'Normal Spread', score: 0.2),
                const Spacer(),
                Text('Organic spread',
                    style: Theme.of(context).textTheme.labelSmall),
              ],
            )),
      ],
    );
  }

  Widget _overviewCard(String title, String subtitle, IconData icon,
      String badgeLabel, double badgeScore, String detail) {
    return GlassCard(title: title, subtitle: subtitle, icon: icon,
        child: Row(
          children: [
            SuspicionBadge(level: badgeLabel, score: badgeScore),
            const Spacer(),
            Text(detail, style: Theme.of(context).textTheme.labelSmall),
          ],
        ));
  }

  Widget _modalitiesContent(bool dark) {
    return Column(
      children: [
        GlassCard(title: 'Modality Importance', icon: Icons.tune,
            child: const ModalityBreakdown(scores: {
              'Text': 0.35, 'Image': 0.25,
              'Credibility': 0.22, 'Propagation': 0.18,
            })),
        const SizedBox(height: HIGSpacing.md),
        GlassCard(title: 'Attention Distribution', icon: Icons.visibility,
            child: Text('Fusion model attention weights across modalities',
                style: Theme.of(context).textTheme.bodyMedium)),
      ],
    );
  }

  Widget _explainContent(bool dark) {
    return Column(
      children: [
        GlassCard(title: 'LIME Explanation',
            subtitle: 'Top textual features driving prediction',
            icon: Icons.search,
            child: Column(
              children: [
                _explainRow('breaking', 0.42, false, dark),
                _explainRow('shocking', 0.38, false, dark),
                _explainRow('allegedly', 0.21, false, dark),
                _explainRow('according', 0.15, true, dark),
              ],
            )),
        const SizedBox(height: HIGSpacing.md),
        GlassCard(title: 'Decision Path', icon: Icons.account_tree,
            child: Column(
              children: [
                _decisionStep('Text Analysis', 0.35, 'High emotional language', dark),
                _decisionStep('Image Verification', 0.25, 'No manipulation', dark),
                _decisionStep('Source Check', 0.22, 'Low credibility', dark),
                _decisionStep('Propagation', 0.18, 'Bot-like cascade', dark),
                const Divider(color: LightColors.separator),
                Center(
                  child: Text('Prediction: Misinformation',
                      style: TextStyle(
                        fontWeight: FontWeight.w600,
                        color: dark ? DarkColors.danger : LightColors.danger,
                      )),
                ),
              ],
            )),
      ],
    );
  }

  Widget _explainRow(String word, double weight, bool positive, bool dark) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 4),
      child: Row(
        children: [
          Text(word, style: Theme.of(context).textTheme.bodyLarge),
          const Spacer(),
          Text(
            '${positive ? '+' : '-'}${weight.toStringAsFixed(2)}',
            style: TextStyle(
              fontSize: 12,
              fontWeight: FontWeight.w600,
              color: positive
                  ? (dark ? DarkColors.tertiary : LightColors.tertiary)
                  : (dark ? DarkColors.danger : LightColors.danger),
            ),
          ),
        ],
      ),
    );
  }

  Widget _decisionStep(String label, double score, String detail, bool dark) {
    final dotColor = score > 0.2
        ? (dark ? DarkColors.primary : LightColors.primary)
        : (dark ? DarkColors.tertiary : LightColors.tertiary);
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 6),
      child: Row(
        children: [
          Container(width: 6, height: 6,
              decoration: BoxDecoration(color: dotColor, shape: BoxShape.circle)),
          const SizedBox(width: 8),
          Text(label, style: Theme.of(context).textTheme.bodyLarge),
          const Spacer(),
          Text(detail, style: Theme.of(context).textTheme.bodyMedium),
        ],
      ),
    );
  }
}

File: mobile/Views/DetailView.swift
// ============================================================
// [Software Eng] Detail Analysis View
// ============================================================
// Per-post breakdown showing text, image, credibility, and
// propagation analysis results with glass cards.

import SwiftUI

struct DetailView: View {
    let post: PostItem
    @State private var selectedSegment = 0
    @State private var showShareSheet = false

    private let segments = ["Overview", "Modalities", "Explain"]

    var body: some View {
        ScrollView {
            VStack(spacing: SpacingHIG.medium) {
                headerSection
                scoreSection
                segmentPicker
                segmentContent
            }
            .padding()
        }
        .background(Color.mgBackground)
        .navigationTitle("Analysis")
        .navigationBarTitleDisplayMode(.inline)
        .toolbar {
            ToolbarItem(placement: .navigationBarTrailing) {
                Button(action: { showShareSheet = true }) {
                    Image(systemName: "square.and.arrow.up")
                        .foregroundColor(.mgPrimary)
                }
            }
        }
    }

    private var headerSection: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(post.title)
                .font(.mgTitle)
                .foregroundColor(.labelPrimary)

            HStack {
                Image(systemName: "clock")
                    .font(.caption)
                    .foregroundColor(.mgTertiary)
                Text(post.timestamp, style: .date)
                    .font(.mgCaption)
                    .foregroundColor(.labelSecondary)

                Spacer()

                Image(systemName: "person.circle")
                    .font(.caption)
                    .foregroundColor(.mgTertiary)
                Text(post.source)
                    .font(.mgCaption)
                    .foregroundColor(.labelSecondary)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }

    private var scoreSection: some View {
        GlassCard(title: "Overall Suspicion Score") {
            VStack(spacing: SpacingHIG.medium) {
                HStack {
                    Spacer()
                    ConfidenceGauge(
                        score: post.suspicionScore,
                        label: post.suspicionScore > 0.7 ? "Likely Misinformation" : "Likely Authentic",
                        size: 120
                    )
                    Spacer()
                }

                ConfidenceBar(
                    score: post.suspicionScore,
                    label: "Confidence"
                )
            }
        }
    }

    private var segmentPicker: some View {
        Picker("View", selection: $selectedSegment) {
            ForEach(0..<segments.count, id: \.self) { i in
                Text(segments[i]).tag(i)
            }
        }
        .pickerStyle(.segmented)
    }

    @ViewBuilder
    private var segmentContent: some View {
        switch selectedSegment {
        case 0: overviewContent
        case 1: modalitiesContent
        case 2: explainContent
        default: EmptyView()
        }
    }

    private var overviewContent: some View {
        VStack(spacing: SpacingHIG.medium) {
            GlassCard(
                title: "Text Analysis",
                subtitle: "Semantic and linguistic patterns analyzed",
                icon: "text.alignleft"
            ) {
                HStack {
                    SuspicionBadge(
                        level: post.suspicionScore > 0.7 ? "Suspicious" : "Normal",
                        score: post.suspicionScore
                    )
                    Spacer()
                    Text("BERT-based encoding")
                        .font(.mgCaption)
                        .foregroundColor(.labelTertiary)
                }
            }

            if post.hasImage {
                GlassCard(
                    title: "Image Analysis",
                    subtitle: "Visual integrity verification via ResNet",
                    icon: "photo"
                ) {
                    HStack {
                        SuspicionBadge(
                            level: "Authentic",
                            score: 0.15
                        )
                        Spacer()
                        Text("No manipulation detected")
                            .font(.mgCaption)
                            .foregroundColor(.labelTertiary)
                    }
                }
            }

            GlassCard(
                title: "Source Credibility",
                subtitle: "Dynamic scoring based on historical behavior",
                icon: "person.badge.shield.checkmark"
            ) {
                ConfidenceBar(score: 0.78, label: "Credibility Score")
            }

            GlassCard(
                title: "Propagation Pattern",
                subtitle: "Share network modeled as graph",
                icon: "point.3.connected.trianglepath.dotted"
            ) {
                HStack {
                    SuspicionBadge(
                        level: post.suspicionScore > 0.7 ? "Suspicious Cascade" : "Normal Spread",
                        score: post.suspicionScore
                    )
                    Spacer()
                    Text("\(post.suspicionScore > 0.7 ? "Bot-like pattern detected" : "Organic spread")")
                        .font(.mgCaption)
                        .foregroundColor(.labelTertiary)
                }
            }
        }
    }

    private var modalitiesContent: some View {
        VStack(spacing: SpacingHIG.medium) {
            GlassCard(title: "Modality Importance", icon: "slider.horizontal.3") {
                ModalityBreakdown(modalityScores: [
                    "Text": 0.35,
                    "Image": 0.25,
                    "Credibility": 0.22,
                    "Propagation": 0.18,
                ])
            }

            GlassCard(title: "Attention Distribution", icon: "eye") {
                Text("Fusion model attention weights across modalities")
                    .font(.mgBody)
                    .foregroundColor(.labelSecondary)

                HStack(spacing: 4) {
                    ForEach(0..<4) { i in
                        RoundedRectangle(cornerRadius: 4)
                            .fill(i == 0 ? .mgPrimary : .mgTertiary)
                            .frame(height: CGFloat(20 - i * 4))
                    }
                }
                .frame(height: 20)
            }
        }
    }

    private var explainContent: some View {
        VStack(spacing: SpacingHIG.medium) {
            GlassCard(
                title: "LIME Explanation",
                subtitle: "Top textual features driving this prediction",
                icon: "doc.text.magnifyingglass"
            ) {
                VStack(alignment: .leading, spacing: 8) {
                    explainRow(word: "breaking", weight: 0.42, positive: false)
                    explainRow(word: "shocking", weight: 0.38, positive: false)
                    explainRow(word: "allegedly", weight: 0.21, positive: false)
                    explainRow(word: "according", weight: 0.15, positive: true)
                }
            }

            GlassCard(
                title: "Grad-CAM Heatmap",
                subtitle: "Image regions most influential to prediction",
                icon: "square.grid.3x3.fill.square"
            ) {
                RoundedRectangle(cornerRadius: 8)
                    .fill(.quaternary)
                    .frame(height: 160)
                    .overlay(
                        Image(systemName: "photo.on.rectangle")
                            .font(.title)
                            .foregroundColor(.mgTertiary)
                    )
            }

            GlassCard(
                title: "Decision Path",
                icon: "arrow.triangle.branch"
            ) {
                VStack(alignment: .leading, spacing: 12) {
                    decisionStep("Text Analysis", score: 0.35, detail: "High emotional language")
                    decisionStep("Image Verification", score: 0.25, detail: "No manipulation flags")
                    decisionStep("Source Check", score: 0.22, detail: "Low credibility")
                    decisionStep("Propagation", score: 0.18, detail: "Bot-like cascade")
                    HStack {
                        Spacer()
                        Text("Prediction: Misinformation")
                            .font(.mgHeadline)
                            .foregroundColor(.mgDanger)
                        Spacer()
                    }
                    .padding(.top, 8)
                }
            }
        }
    }

    private func explainRow(word: String, weight: Double, positive: Bool) -> some View {
        HStack {
            Text(word)
                .font(.mgBody)
                .foregroundColor(.labelPrimary)

            Spacer()

            Text("\(positive ? "+" : "-")\(String(format: "%.2f", weight))")
                .font(.mgCaption.weight(.semibold))
                .foregroundColor(positive ? .mgTertiary : .mgDanger)
        }
        .padding(.horizontal, 8)
        .padding(.vertical, 4)
        .background((positive ? Color.mgTertiary : Color.mgDanger).opacity(0.05))
        .clipShape(RoundedRectangle(cornerRadius: 6))
    }

    private func decisionStep(_ label: String, score: Double, detail: String) -> some View {
        HStack {
            Circle()
                .fill(score > 0.2 ? .mgPrimary : .mgTertiary)
                .frame(width: 6, height: 6)

            Text(label)
                .font(.mgBody)
                .foregroundColor(.labelPrimary)

            Spacer()

            Text(detail)
                .font(.mgCaption)
                .foregroundColor(.labelSecondary)
        }
    }
}
